test/auto-languageclient.test.ts

```diff
@@ -12,20 +19,91 @@ function mockEditor(uri: string, scopeName: string): any {
   }
 }
 
+function setupClient() {
+  atom.workspace.getTextEditors().forEach((editor) => editor.destroy())
+  atom.project.getPaths().forEach((project) => atom.project.removePath(project))
+  const client = new FakeAutoLanguageClient()
+  client.activate()
+  return client
+}
+
+function setupServerManager(client = setupClient()) {
+  /* eslint-disable-next-line dot-notation */
+  const serverManager = client["_serverManager"]
+  return serverManager
+}
+
 describe("AutoLanguageClient", () => {
+  describe("determineProjectPath", () => {
+    it("returns null when a single file is open", async () => {
+      const client = setupClient()
+      const textEditor = (await atom.workspace.open(__filename)) as TextEditor
+      /* eslint-disable-next-line dot-notation */
+      const projectPath = client["determineProjectPath"](textEditor)
+      expect(projectPath).toBeNull()
+    })
+    it("returns the project path when a file of that project is open", async () => {
+      // macos has issues with handling too much resources
```

Comment:
is this just an issue on GitHub actions or would this crash Atom if run on a Mac?

Reply:
Because the number of tests is high, MacOS can't run all of them. It works fine if each of the tests runs in a single process. There is nothing OS-specific in the code, so this is a limitation of the operating system.